hydromt/models/model.py

```diff
@@ -133,6 +130,38 @@ def __init__(
             f"Initializing {self._NAME} model from {model_metadata['plugin_name']} (v{model_metadata['version']})."
         )
 
+        self._region_component_name = self._determine_region_component(region_component)
+
+    def _determine_region_component(self, region_component: Optional[str]) -> str:
+        if region_component is not None:
+            if region_component not in self._components:
+                self.logger.warning(
+                    f"Component {region_component} not found in components."
+                    "You can add it afterwards with add_component."
+                )
+            elif not isinstance(
+                self._components.get(region_component, None), SpatialModelComponent
+            ):
+                raise ValueError(
+                    f"Component {region_component} is not a {SpatialModelComponent.__name__}."
+                )
+            return region_component
+        else:
+            has_region_components = [
+                (name, c)
+                for name, c in self._components.items()
+                if isinstance(c, SpatialModelComponent)
+            ]
+            if len(has_region_components) > 1:
+                raise ValueError(
+                    "Multiple region components found in components. "
+                    "Specify region_component."
+                )
```

Comment:
Maybe plugins will want to overwrite this (eg model region is actually the total bounds of several components). Then the method should be public and documented.

Reply:
No, let's implement a good component that can for example:

```yaml
global:
  region_component: region
  components:
    region:
      type: CompoundRegionComponent
      regions:
        - grid
        - vector
        - subgrid
    grid:
      type: GridComponent
    # etc
```

And implement it like so:

```py
class CompoundRegionComponent(SpatialModelComponent):
  def __init__(model, regions):
    pass

  def region(self):
    # pseudo-code
    expand([r.bounds for r in self.regions])
```

Reply:
We will get conflicts during loading. @savente93 made error handling for this, but it's not possible to load two `GridComponent`s in at the moment.